suite/tests/client-interface/drx-scattergather.dll.c

```diff
@@ -63,24 +65,59 @@ inscount(uint num_instrs)
     global_sg_count += num_instrs;
 }
 
+static bool one_time_clobber_test = false;
+
 static dr_emit_flags_t
 event_app_instruction(void *drcontext, void *tag, instrlist_t *bb, instr_t *instr,
                       bool for_trace, bool translating, void *user_data)
 {
     uint num_instrs;
-    if (!drmgr_is_first_instr(drcontext, instr))
-        return DR_EMIT_DEFAULT;
+    if (one_time_clobber_test) {
+        if (instr_get_opcode(instr) == OP_kandnw) {
+            instr_t *search_app_instr = instr;
+            while (!instr_is_app(search_app_instr)) {
+                search_app_instr = instr_get_next(search_app_instr);
+            }
+            /* We chose to do this in the insertion phase, even though we're inserting an
+             * app instruction. We're finding the clobber case of the scatter/gather
```

Comment:
I don't see the argument for *why* this is not done in the app2app phase?

Reply:
Then I don't need to scan all app lists every time for the markers, because I am getting called on every app instruction, it was more convenient. I've added a comment.

Reply:
Is this important to you so I need to change or good as-is after comment update?

Reply:
You mean this is just to avoid an outer loop?  That does not seem enough to violate the phase rules.

Reply:
Moved to app2app phase.